src/data-access/repositories/mongo/mongo-device-changed-listener.ts

```diff
@@ -0,0 +1,93 @@
+import { DataChangedListener } from '../interfaces/data-changed-listener'
+import { MongoDevice } from './mongo-entity-converter'
+import { BaseMongoRepository } from './base-mongo-repository'
+import { MongoDatabase } from './mongo-database'
+import { ChangeStream, ChangeStreamDocument, ChangeStreamOptions, ChangeStreamUpdateDocument } from 'mongodb'
+import { MongoChangeEvent } from './mongo-enums'
+import { Device } from '../../../model/entities/device'
+import { StatusCode } from '../../../model/enums/status-code'
+import { Logger } from '../../../logger/logger'
+import { UnsupportedOperationException } from '../../../model/exceptions/unsupported-operation-exception'
+
+const DEVICE_COLLECTION_NAME: string = 'peripheralDevices'
+
+export class MongoDeviceChangedListener extends BaseMongoRepository implements DataChangedListener<Device> {
+
+  private readonly logger: Logger
+  private onUpdatedCallback: (device: Device) => void
+
+  constructor(mongoDatabase: MongoDatabase, logger: Logger) {
+    super(mongoDatabase)
+    this.logger = logger.tag(MongoDeviceChangedListener.name)
+    mongoDatabase.onConnect(DEVICE_COLLECTION_NAME, () => this.listenForChanges())
+  }
+
+  private listenForChanges(): void {
+    const options: ChangeStreamOptions = { fullDocument: 'updateLookup' }
+    const changeStream: ChangeStream = this.getCollection().watch<MongoDevice, ChangeStreamDocument<MongoDevice>>([], options)
+    changeStream.on('change', (change: ChangeStreamDocument<MongoDevice>) => void this.onChange(change))
+    this.logger.debug('Listening for Device collection changes...')
+  }
+
+  private onChange(change: ChangeStreamDocument<MongoDevice>): void {
+    switch (change.operationType) {
+      case MongoChangeEvent.UPDATE: {
+        const updateChange: ChangeStreamUpdateDocument<MongoDevice> = change as ChangeStreamUpdateDocument<MongoDevice>
+        const mongoDevice: MongoDevice | undefined = updateChange.fullDocument
+        if (!mongoDevice) {
+          return
+        }
+        this.onUpdatedCallback(this.convertMongoDeviceToDevice(mongoDevice))
+        return
+      }
+    }
+  }
```

Comment:
Here I get it we listen for the changes and react it to change, but if I don't see the model from the names on to the methods I can't get it what we expect to change and what is react to that change. Suggestions for names: `listenForStreamDocumentChanges `and `onStreamDocumentChange`